middleware.go

```diff
@@ -80,7 +83,7 @@ var (
 	}
 )
 
-func SetupMiddleware(api *rest.Api, mwtype string) error {
+func SetupMiddleware(api *rest.Api, mwtype string, privkey *rsa.PrivateKey) error {
```

Comment:
Same here, instead of passing the key around, I would rather pass `Authorizer` and let the caller decide whether to use a simple authz or something else.